dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -611,10 +797,16 @@ private static (Type ReturnType, Func<Kernel, KernelFunction, object?, ValueTask
             }
 
             // ValueTask<T>
-            if (returnType.GetGenericTypeDefinition() is Type genericValueTask &&
-                genericValueTask == typeof(ValueTask<>) &&
-                returnType.GetMethod("AsTask", BindingFlags.Public | BindingFlags.Instance) is MethodInfo valueTaskAsTask &&
-                valueTaskAsTask.ReturnType.GetProperty("Result", BindingFlags.Public | BindingFlags.Instance)?.GetGetMethod() is MethodInfo asTaskResultGetter)
+#if NET6_0_OR_GREATER
+            if (returnType.GetGenericTypeDefinition() is Type genericValueTask && genericValueTask == typeof(ValueTask<>) &&
```

Comment:
```suggestion
            if (returnType.GetGenericTypeDefinition() == typeof(ValueTask<>) &&
```

I don't think we need an intermediate variable here.

Reply:
Yep, thanks, removed.